pilot/pkg/networking/core/cluster_builder.go

```diff
@@ -48,6 +49,21 @@ import (
 
 var maxSecondsValue = int64((math.MaxInt64 - 999999999) / (1000 * 1000 * 1000)) // 9223372035, which is about 292 years.
 
+// See `extensions.network.dns_resolver.cares.v3.CaresDnsResolverConfig` in
+// https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/network/dns_resolver/cares/v3/cares_dns_resolver.proto,
+// `ARES_OPT_UDP_MAX_QUERIES` in https://c-ares.org/docs/ares_init.html, and discussion in https://github.com/c-ares/c-ares/issues/913
+// This value defaults to 0 which is interpreted as infinity and causes Envoys to get "stuck" to dead dns pods
+// (https://github.com/istio/istio/issues/53577).
+// However, setting this value too high will rate limit the number of DNS requests we can make by using up all available local udp ports.
+// Generally, the max dns query rate is
+//
+//	(# local dns ports) * (udp_max_queries) / (query duration)
+//
+// The longest a query can take should be 5s, the Envoy default timeout.
+// We underestimate the number of local dns ports to 10,000 (default is ~15,000, but some might be in use).
+// Setting udp_max_queries to 100 gives us at least 10,000 * 100 / 5 = 200,000 requests / second, which is hopefully enough for any cluster.
+var caresDNSUDPMaxQueries = uint32(100)
```

Comment:
Can we have an env var as well, defaulting to this? That way we if this causes issues, users can turn it off or tune it. Then we can take user feedback and either change defaults, remove it, or promote to a proper API